ciri/api/api.c

```diff
@@ -7,42 +7,212 @@
 
 #include <string.h>
 
+#include "cclient/serialization/json/json_serializer.h"
 #include "ciri/api/api.h"
+#include "request/requests.h"
+#include "response/responses.h"
 #include "utils/logger_helper.h"
 
 #define API_LOGGER_ID "api"
 
+/*
+ * Private functions
+ */
+
+static retcode_t get_node_info(get_node_info_res_t *const res) { return RC_OK; }
+
+static retcode_t get_neighbors(get_neighbors_res_t *const res) { return RC_OK; }
+
+static retcode_t add_neighbors(add_neighbors_req_t const *const req,
+                               add_neighbors_res_t *const res) {
+  return RC_OK;
+}
+
+static retcode_t remove_neighbors(remove_neighbors_req_t const *const req,
+                                  remove_neighbors_res_t *const res) {
+  return RC_OK;
+}
+
+static retcode_t get_tips(get_tips_res_t *const res) { return RC_OK; }
+
+static retcode_t find_transactions(find_transactions_req_t const *const req,
+                                   find_transactions_res_t *const res) {
+  return RC_OK;
+}
+
+static retcode_t get_trytes(get_trytes_req_t const *const req,
+                            get_trytes_res_t *const res) {
+  return RC_OK;
+}
+
+static retcode_t get_inclusion_states(
+    get_inclusion_state_req_t const *const req,
+    get_inclusion_state_res_t *const res) {
+  return RC_OK;
+}
+
+static retcode_t get_balances(get_balances_req_t const *const req,
+                              get_balances_res_t *const res) {
+  return RC_OK;
+}
+
+static retcode_t get_transactions_to_approve(
+    get_transactions_to_approve_req_t const *const req,
+    get_transactions_to_approve_res_t *const res) {
+  return RC_OK;
+}
+
+static retcode_t attach_to_tangle(attach_to_tangle_req_t const *const req,
```

Comment:
These methods should be able to be called directly.
- tests need this
- a custom IRI node with user code might want to call them without going through the API overhead

Reply:
Done.